platform/pulpcore/app/models/content.py

```diff
@@ -63,11 +92,33 @@ def storage_path(self, name):
         """
         return default_storage.get_artifact_path(self.sha256)
 
-    file = models.FileField(db_index=True, upload_to=storage_path, max_length=255)
-    size = models.IntegerField(blank=False, null=False)
-    md5 = models.CharField(max_length=32, blank=False, null=False, unique=False, db_index=True)
-    sha1 = models.CharField(max_length=40, blank=False, null=False, unique=False, db_index=True)
-    sha224 = models.CharField(max_length=56, blank=False, null=False, unique=False, db_index=True)
-    sha256 = models.CharField(max_length=64, blank=False, null=False, unique=True, db_index=True)
-    sha384 = models.CharField(max_length=96, blank=False, null=False, unique=True, db_index=True)
-    sha512 = models.CharField(max_length=128, blank=False, null=False, unique=True, db_index=True)
+    artifact = models.ForeignKey(Artifact, on_delete=models.CASCADE, null=True)
+    content = models.ForeignKey(Content, on_delete=models.CASCADE)
+    relative_path = models.CharField(max_length=64)
+
+
+class DeferredArtifact(Model):
+    """
+    A URL for fetching a file associated with a piece of content.
+
+    Fields:
+
+        url (models.TextField): The URL where the artifact can be retrieved.
+        size (models.IntegerField): The expected size of the file in bytes.
+        md5 (models.CharField): The expected MD5 checksum of the file.
+        sha1 (models.CharField): The expected SHA-1 checksum of the file.
+        sha224 (models.CharField): The expected SHA-224 checksum of the file.
+        sha256 (models.CharField): The expected SHA-256 checksum of the file.
+        sha384 (models.CharField): The expected SHA-384 checksum of the file.
+        sha512 (models.CharField): The expected SHA-512 checksum of the file.
+    """
+    url = models.TextField(blank=True, validators=[validators.URLValidator])
```

Comment:
Looking at some docs, django stores a CharField as a VARCHAR, and postgres treats a non-size-limited VARCHAR field the same as a TEXT field. So we should get the same behavior and performance out of a ``CharField`` as a ``TextField``.